doc-deploy-dev/action.yml

```diff
@@ -134,9 +134,42 @@ runs:
       uses: actions/checkout@v4
       with:
         repository: ${{ env.REPOSITORY }}
-        ref: ${{ inputs.branch }}
         token: ${{ inputs.token }}
 
+    - name: "Check if branch exists"
+      env:
+        BRANCH: ${{ inputs.branch }}
+      shell: bash
+      run: |
+        # Check the gh-pages branch exists on remote
+        branch_exists=$(git ls-remote --heads origin refs/heads/${{ env.BRANCH }} 2>&1)
+
+        # If the ${{ env.BRANCH }} doesn't exist, then print error message and exit 1
+        if [ -z "$branch_exists" ]; then
+          echo "The ${{ env.BRANCH }} branch does not exist. Creating ${{ env.BRANCH }}."
+
+          # Create orphan branch
+          git checkout --orphan ${{ env.BRANCH }}
+
+          # Remove all files in staging area
+          git rm -rf .
+
+          # Create .nojekyll file
+          touch .nojekyll
+
+          # Configure git username & email
+          git config user.name 'pyansys-ci-bot'
+          git config user.email 'pyansys.github.bot@ansys.com'
+
```

Comment:
Can't it just do it in behalf of the existing token's user?

Reply:
I got this error message when I removed this from the action: 

```
Author identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'runner@...)
```

Reply:
@MaxJPRey Sure, I don't mind those values being secrets. If we change them to secrets the values will still be in the commit history though

Reply:
I wouldn't bother honestly. Info is still there like @klmcadams points out. So there is no real added value.